src/app/models/book-model.ts

```diff
@@ -0,0 +1,30 @@
+import { bookType } from '../book-type.enum';
+import { BookBorrow } from './book-borrow';
+import { Author } from './author-model';
+import { Guid } from 'guid-typescript';
+
+export class Book {
+  public id: Guid;
+  public name: string;
+  public bookCategory: bookType;
+  public author: Author;
+  public price: number;
+  public rentalPrice: number;
+  public borrowData: BookBorrow;
+
+  public constructor(
+    id: Guid,
```

Comment:
Now when you have id as guid you don't need to get id on object construction. Id can be removed from constructor arguments